### PR TITLE
SerpApi GET Method Implementation

### LINKED ISSUE #112: Setting up Github Actions
### Issue Description

Github Actions should be configured to automate testing.

### Steps

- [x] PR created
- [ ] PR reviewed
- [ ] PR merged

### Deadline for the Issue

_No response_

### PR DESCRIPTION
**Pull request by hakanaktas0**
This PR includes the implementation of the GET method that utilizes the SerpApi. It also includes the yaml file of Github Actions for automated testing.

Fixes #112

### CODE REVIEW

File: practice_app/api/views.py [hunk 1]
```diff
@@ -1,3 +1,71 @@
-from django.shortcuts import render
+import requests
+import re
+from django.http import JsonResponse
+from . import api_keys
 
-# Create your views here.
+
+def google_scholar(request):
+    url = request.get_full_path().split('?')[-1] + '&'
```
Comment (defabdullah) [inline practice_app/api/views.py:8]: You can use request.GET to get dictionary

Comment (hakanaktas0) [reply practice_app/api/views.py]: It is fixed

File: practice_app/api/views.py [hunk 2]
```diff
@@ -1,3 +1,66 @@
...
+        request = request.json()
+        papers = request['organic_results']
+        response = {}
+        results = []
+        for paper in papers:
+            pub_info = paper['publication_info']
+            paper_info = {}
+            paper_info['source'] = 'google_scholar'
+
+            if 'authors' in  pub_info.keys():
```
Comment (defabdullah) [inline practice_app/api/views.py:29]: I suppose we decided to remove author field

Comment (hakanaktas0) [reply practice_app/api/views.py:29]: We decided to decide it later.

**Review by mehmetSuzer — APPROVED** [review]
Good job, I think it is a great start for the implementation of our web application.

**Comment by hakanaktas0** [comment]
All reviews are completed. I am merging this PR.

File: practice_app/api/tests.py [hunk 3]
```diff
@@ -1,3 +1,50 @@
-from django.test import TestCase
-
+from django.test import TestCase,Client
+from unittest import skip
+import requests
+from . import api_keys
 # Create your tests here.
+class google_scholar_test_cases(TestCase):
```
Comment (laylaylo) [inline practice_app/api/tests.py:6]: Hey, I was just looking at this code again and I noticed that the class naming doesn't quite align with our decision to use PEP-8 Style [class naming convention](https://peps.python.org/pep-0008/#class-names). Just a friendly reminder for future reference!

Comment (hakanaktas0) [reply practice_app/api/tests.py:6]: Thx for the heads up. I will fix them in a future PR.

**Pull request merged**